app/src/main/java/org/schabi/newpipe/ExitActivity.java

```diff
@@ -48,6 +48,8 @@ protected void onCreate(final Bundle savedInstanceState) {
             finish();
         }
 
-        System.exit(0);
+        finishAffinity();
+        final Intent intent = new Intent(this, MainActivity.class);
+        startActivity(intent);
```

Comment:
Create a common `restartApp(final Activity activity)` method for this in the `util/NavigationHelper` class and reuse it in both places

Reply:
I've created the common method and updated ExitActivity.java and ContentSettingsFragment.java :), changes have been pushed to my branch